valohai_cli/utils/ssh.py

```diff
@@ -0,0 +1,131 @@
+import json
+import os
+import pathlib
+import shutil
+import subprocess
+from dataclasses import dataclass
+from typing import Generator, Iterator, Optional, Tuple
+
+import click
+
+from valohai_cli.api import request
+from valohai_cli.ctx import get_project
+from valohai_cli.settings import get_settings_file_name
+from valohai_cli.utils.cli_utils import prompt_from_list
+
+
+@dataclass
```

Comment:
Might as well make it immutable:
```suggestion
@dataclass(frozen=True)
```